specs/commandline/sign.md

```diff
@@ -4,7 +4,21 @@
 
 Use `notation sign` to sign artifacts.
 
-Signs an OCI artifact that is stored in a registry. Upon successful signing, the generated signature is pushed to the registry with the digest of the OCI artifact returned.
+Signs an OCI artifact that is stored in a registry. Always use a `digest` to identify an artifact. `Tags` are mutable, but `digests` uniquely and immutably identify artifacts. If a tag is used, notation resolves the tag to the `digest` before signing.
+
+Upon successful signing, the generated signature is pushed to the registry and associated with the signed OCI artifact. The output message is printed out as following:
+
+```text
+Signing of artifact <reference> succeeded and the signature is attached to <registry>/<repository>@<digest>.
+```
+
+If a `tag` is used to identify the OCI artifact, the output message is as following:
+
+```test
+Warning: A tag is used to identify the artifact for signing. Artifact tags are mutable. Use digests to uniquely identify artifacts and avoid mutability.
+Resolving artifact tag '<tag>' to digest '<digest>' before signing.
+Signing of artifact <reference> succeeded and the signature is attached to <registry>/<repository>@<digest>.
```

Comment:
```suggestion
Warning: Always sign artifact using digest(`@sha256:...`) rather than a tag(`:latest`) because tags are mutable and you might end up signing something that you didnt intend to.
Successfully signed <registry>/<repository>@<digest>
```

Reply:
Suggest using the following warning message:

"Warning: Always sign the artifact using digest(`@sha256:...`) rather than a tag(`:latest`) because tags are mutable and a tag reference can point to a different artifact than the one signed."